### PR TITLE
Don't use ifdefs for our Dev14/Dev15 completion impl.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Instead using a facade/mixin approach to more cleanly pull this off.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide 
@Pilchie This is what i was thinking about for completion.  No more ifdefs.  However, it does mean a fair bit of boilerplate forwarding code in the impls of VisualStudio14CompletionSet and VisualStudio15CompletionSet.  However, as it's all just boilerplate, i don't mind too much.  Plus, i'm much happier to not have anymore ifdefs in this code.

**Comment by CyrusNajmabadi** [comment]
@Pilchie can you please unpack on your Dev15 box and see if things still work.

Note: i don't have your fix merged in i dn't think.  So empty filtered lists will still crash i think.

**Comment by CyrusNajmabadi** [comment]
@Pilchie  Please review this.  This will affect the work i'm doing to not pull in dev15 binaries if no options are enabled.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/IVsCompletionSet.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    /// <summary>
+    /// See comment on VisualStudio15CompletionSet for an explanation of how these types 
+    /// fit together and where code should go in them.
+    /// 
+    /// This type exists so we can expose normally protected properties on <see cref="CompletionSet"/> 
+    /// to types like <see cref="Roslyn14CompletionSet"/> so they can read/write them.  This allows
+    /// us to encapulate logic in our own inheritance hierarchy without having to fit into the
+    /// editor's inheritance hierarchy.
+    /// </summary>
+    internal interface IVsCompletionSet : ICompletionSet
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/IVsCompletionSet.cs]: Please do not call this `IVsCompletionSet` - and `IVs` prefix makes me think of the COM based VS extensibility model for VS, not something that we built ourselves.

Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/IVsCompletionSet.cs]: I'd rather `IVisualStudioCompletionSet` to fit in with the other names.

**Comment by Pilchie** [comment]
Also - please remove the `Dev15` preprocessor symbol from the `EditorFeatures.Next.csproj` as well.

**Pull request merged**